uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/BAO/LeaveRequest.php

```diff
@@ -65,6 +65,38 @@ private static function validateParams($params) {
       );
     }
 
+    if (empty($params['contact_id'])) {
+      throw new InvalidLeaveRequestException(
+        'Leave Request should have a contact',
+        'leave_request_empty_contact_id',
+        'contact_id'
+      );
+    }
+
+    if (empty($params['type_id'])) {
+      throw new InvalidLeaveRequestException(
+        'Leave Request should have an Absence Type',
+        'leave_request_empty_type_id',
+        'type_id'
+      );
+    }
+
+    if (empty($params['status_id'])) {
+      throw new InvalidLeaveRequestException(
+        'The Leave Request status should not be empty',
+        'leave_request_empty_status_id',
+        'status_id'
+      );
+    }
+
+    if (!empty($params['to_date']) && empty($params['to_date_type'])) {
+      throw new InvalidLeaveRequestException(
+        'The type of To Date should not be empty',
+        'leave_request_empty_to_date_type',
+        'to_date_type'
+      );
+    }
+
```

Comment:
This method got quite big now with all those new rules. Can you please refactor it and put the validations into separate methods? You can group them by type of validation. For example, you can group all the validations for mandatory fields into a `validateMandatory()` method